### PR TITLE
fix: ignore accessible source properties with inaccessible getters and accessible target properties with inaccessible getters

### LINKED ISSUE #140: "The property or indexer cannot be used in this context because the set accessor is inaccessible"
**Describe the bug**
Properties with private/protected setters throw exception

**To Reproduce**
```csharp
public class Foo
{
    public string Bar { get; private set; } //see, the property is public but the setter is private
}
```

**Expected behavior**
I suggest skipping properties with non-public setters

### PR DESCRIPTION
**Pull request by latonz**
Closes #140

### CODE REVIEW

File: test/Riok.Mapperly.Tests/Mapping/ObjectPropertyTest.cs [hunk 1]
```diff
@@ -303,4 +303,56 @@ public Task WithNotFoundIgnoredSourcePropertyShouldDiagnostic()
...
+    [Fact]
+    public Task WithPrivateSourcePathGetterShouldIgnoreAndDiagnostic()
+    {
+        var source = TestSourceBuilder.Mapping(
+            "A",
+            "B",
+            "class A { public C StringValue { private get; set; } public int IntValue { get; private set; } }",
+            "class B { public C StringValue { get; set; } public int IntValue { private get; set; } }",
+            "class C { public string StringValue { get; set; } }",
+            "class D { public string StringValue { get; set; } }");
```
Comment (CommonGuy) [inline test/Riok.Mapperly.Tests/Mapping/ObjectPropertyTest.cs:354]: Class `D` is unused here. In all the new test cases, I would use a name other than `StringValue` for the properties of type `C` and `D`. For example `Nested` or something else

Comment (latonz) [reply test/Riok.Mapperly.Tests/Mapping/ObjectPropertyTest.cs:354]: Improved.

Comment (CommonGuy) [reply test/Riok.Mapperly.Tests/Mapping/ObjectPropertyTest.cs:354]: Now class `C` is unused in all tests

Comment (latonz) [reply test/Riok.Mapperly.Tests/Mapping/ObjectPropertyTest.cs:354]: 🤦 should be correct now.

**Review by CommonGuy — APPROVED** [review]

**Pull request merged**